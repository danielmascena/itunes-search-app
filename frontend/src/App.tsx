import React from 'react';
import { useSelector, shallowEqual, useDispatch } from "react-redux";
import { Dispatch } from "redux";

import Home from './pages/Home';

import './App.css';

const App: React.FC = () => {

  const dispatch: Dispatch<any> = useDispatch();

  const resources: readonly IMedia[] = useSelector(
    (state: AppState) => state.mediaCollection,
    shallowEqual
  );

  return (
    <Home />
  );
}

export default App;
