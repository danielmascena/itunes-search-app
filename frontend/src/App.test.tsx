import * as React from "react";
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import App from "./App";

const mockStore = configureStore([]);

test("should renders the App component", () => {
    const store = mockStore({
      search: {
        term: "",
        type: "ALL"
      },
      mediaCollection: []
    });

    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    );
  expect(component.toJSON()).toMatchSnapshot();
});
