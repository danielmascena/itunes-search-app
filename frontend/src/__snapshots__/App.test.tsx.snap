// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders the App component 1`] = `
Array [
  <header>
    <h1>
      Header
    </h1>
  </header>,
  <main>
    <h1>
      iTunes Search App
    </h1>
  </main>,
  <footer>
    <p>
      Copyright
    </p>
  </footer>,
]
`;
